File: app/src/main/java/br/com/medeve/Interfaces/IUsuarioControler.java
package br.com.medeve.Interfaces;

import android.app.Activity;

import br.com.medeve.Models.Cliente;
import br.com.medeve.Models.Usuario;

public interface IUsuarioControler {

    void entrar(Usuario usuario);

    void cadastrar(Usuario usuario, Activity activity);

    void recuperarSenha(Usuario usuario);

    void persistirUsuario(Activity activity, Class clazz);

    void sair(Activity activity, Class clazz);
}

File: app/src/main/java/br/com/medeve/Interfaces/IUsuarioControler.kt
package br.com.medeve.Interfaces

import android.app.Activity
import br.com.medeve.Models.Usuario

interface IUsuarioControler {
    fun entrar(usuario: Usuario?)
    fun cadastrar(usuario: Usuario?, activity: Activity?)
    fun recuperarSenha(usuario: Usuario?)
    fun persistirUsuario(activity: Activity?, clazz: Class<*>?)
    fun sair(activity: Activity?, clazz: Class<*>?)
}